coders/fits.c

```diff
@@ -145,27 +145,34 @@ static MagickBooleanType IsFITS(const unsigned char *magick,const size_t length)
 
 static inline double GetFITSPixel(Image *image,int bits_per_pixel)
 {
+  double pixel = 0;
   switch (image->depth >> 3)
   {
     case 1:
-      return((double) ReadBlobByte(image));
+      pixel=((double) ReadBlobByte(image));
     case 2:
-      return((double) ((short) ReadBlobShort(image)));
+      pixel=((double) ((short) ReadBlobShort(image)));
     case 4:
     {
       if (bits_per_pixel > 0)
-        return((double) ReadBlobSignedLong(image));
-      return((double) ReadBlobFloat(image));
+        pixel=((double) ReadBlobSignedLong(image));
+      pixel=((double) ReadBlobFloat(image));
     }
     case 8:
     {
       if (bits_per_pixel > 0)
-        return((double) ((MagickOffsetType) ReadBlobLongLong(image)));
+        pixel=((double) ((MagickOffsetType) ReadBlobLongLong(image)));
     }
     default:
       break;
   }
-  return(ReadBlobDouble(image));
+  pixel=(ReadBlobDouble(image));
```

Comment:
You are now overwriting the `pixel` value that was set in the switch statement?

Reply:
🤦🏽‍♂️ I got excited after figuring out the NaN thing and forgot how to code.

Reply:
yea, changed the previous case statements to keep their returns, so pixel is only used for ReadBlobDouble here and nothing is overwritten.